gnomad_qc/v3/resources/sample_qc.py

```diff
@@ -1,69 +1,112 @@
-from gnomad.resources import TableResource, MatrixTableResource
-from hail import Table
+import hail as hl
+from gnomad.resources.resource_utils import (MatrixTableResource,
+                                             TableResource,
+                                             VersionedMatrixTableResource,
+                                             VersionedTableResource)
 from gnomad.sample_qc.relatedness import get_relationship_expr
 
-SAMPLE_QC_ROOT = "gs://gnomad/sample_qc/ht/genomes_v3"
+from gnomad_qc.v3.resources.constants import CURRENT_RELEASE, RELEASES
 
 
-def get_sample_qc(strat: str = "all") -> TableResource:
+def get_sample_qc_root(version: str = CURRENT_RELEASE, mt: bool = False) -> str:
+    """
+    Return path to sample QC root folder
+
+    :param version: Version of sample QC path to return
+    :param mt: Whether path is for a MatrixTable, default is False
+    :return: Root to sample QC path
+    """
+    return f"gs://gnomad/sample_qc/{'mt' if mt else 'ht'}/genomes_v{version}"
+
+
+def get_sample_qc(strat: str = "all") -> VersionedTableResource:
     """
     Gets sample QC annotations generated by Hail for the specified stratification:
         - bi-allelic
         - multi-allelic
         - all
 
     :param strat: Which stratification to return
-    :return: sample QC table
+    :return: Sample QC table
     """
-    return TableResource(f'{SAMPLE_QC_ROOT}/sample_qc_{strat}.ht')
+    return VersionedTableResource(
+        CURRENT_RELEASE,
+        {release: TableResource(f"{get_sample_qc_root(release)}/sample_qc_{strat}.ht") for release in RELEASES})
+
 
+def _get_ancestry_pca_ht_path(
+    part: str,
+    version: str = CURRENT_RELEASE,
+    include_unreleasable_samples: bool = False,
+) -> str:
+    """
+    Helper function to get path to files related to ancestry PCA
 
-def _get_ancestry_pca_ht_path(part: str, include_unreleasable_samples: bool = False) -> str:
-    return '{}/gnomad_v3_pca_{}{}.{}'.format(
-        SAMPLE_QC_ROOT,
+    :param part: String indicating the type of PCA file to return (loadings, eigenvalues, or scores)
+    :param version: Version of sample QC path to return
+    :param include_unreleasable_samples: Whether the file includes PCA info for unreleasable samples
+    :return: Path to requested ancestry PCA file
+    """
+    return "{}/gnomad_v{}_pca_{}{}.{}".format(
+        get_sample_qc_root(version),
+        version,
         part,
-        '_with_unreleasable_samples' if include_unreleasable_samples else '',
-        'txt' if part == 'eigenvalues' else 'ht'
+        "_with_unreleasable_samples" if include_unreleasable_samples else "",
+        "txt" if part == "eigenvalues" else "ht",
     )
 
 
-def get_ancestry_pca_loadings(include_unreleasable_samples: bool = False) -> TableResource:
+def ancestry_pca_loadings(include_unreleasable_samples: bool = False
+) -> TableResource:
```

Comment:
This returns a `VersionedTableResource` but is annotated as returning `TableResource`. It's not wrong, a `VersionedTableResource` is a `TableResource`.